packages/thirdweb/src/utils/any-evm/zksync/isZkSyncChain.ts

```diff
@@ -1,12 +1,27 @@
 import type { Chain } from "../../../chains/types.js";
 
-export function isZkSyncChain(chain: Chain) {
-  return (
-    chain.id === 324 ||
-    chain.id === 300 ||
-    chain.id === 302 ||
-    chain.id === 11124 ||
-    chain.id === 282 || // cronos zkevm testnet
-    chain.id === 388 // cronos zkevm mainnet
-  );
+const chainStackCache = new Map();
+
+export async function isZkSyncChain(chain: Chain) {
+  const stack = await getChainStack(chain.id);
+
+  return stack === "zksync-stack";
+}
+
+export async function getChainStack(chainId: number): Promise<string> {
+  if (chainStackCache.has(chainId)) {
+    return chainStackCache.get(chainId);
+  }
+
+  const res = await fetch(`https://${chainId}.rpc.thirdweb.com/stack`);
+  const data = await res.json();
+
+  if (res.ok) {
+    const stack = data.stack;
+    chainStackCache.set(chainId, { stack });
+
+    return stack;
+  } else {
+    throw new Error(`Error: ${res.status} - ${data}`);
+  }
```

Comment:
we have a way to cache a promise: `withCache` use it for this

Reply:
updated